src/__swaps__/screens/Swap/components/PriceImpactWarning.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import * as i18n from '@/languages';
+import Animated, { useDerivedValue } from 'react-native-reanimated';
+import { AnimatedText, Box, Inline, Text, TextIcon, useForegroundColor } from '@/design-system';
+import { opacity } from '@/__swaps__/utils/swaps';
+import { useSwapContext } from '@/__swaps__/screens/Swap/providers/swap-provider';
+import { SwapPriceImpactType } from '@/hooks/usePriceImpactDetails';
+
+export const PriceImpactWarning = () => {
+  const { AnimatedSwapStyles, PriceImpactWarning } = useSwapContext();
+
+  // TODO: i18n is not liking this...
+  const warningText = useDerivedValue(() => {
+    if (PriceImpactWarning.value.type === SwapPriceImpactType.none) return '';
+    return i18n.t(i18n.l.exchange.price_impact.you_are_losing, {
+      impactDisplay: PriceImpactWarning.value.impactDisplay,
+    });
+  });
+
+  const fillTertiary = useForegroundColor('fillTertiary');
+
+  return (
+    <Box
+      as={Animated.View}
+      alignItems="center"
+      justifyContent="center"
+      paddingHorizontal="24px"
+      paddingVertical="12px"
+      style={[AnimatedSwapStyles.hideWhenInputsExpanded, AnimatedSwapStyles.hideWhenPriceWarningIsNotPresent, { alignSelf: 'center' }]}
+    >
+      <Box as={Animated.View} alignItems="center" height={{ custom: 33 }} gap={6} justifyContent="center" paddingHorizontal="10px">
+        <Inline alignHorizontal="center" alignVertical="center" horizontalSpace="4px" wrap={false}>
+          <Box
+            borderRadius={10}
+            height={{ custom: 20 }}
+            paddingTop={{ custom: 0.25 }}
+            style={{ backgroundColor: opacity(fillTertiary, 0.04) }}
+            width={{ custom: 20 }}
+          >
+            <TextIcon color="orange" containerSize={20} size="15pt" weight="heavy">
+              􀇿
+            </TextIcon>
+          </Box>
+          <AnimatedText align="center" color="orange" size="15pt" weight="heavy" text={warningText} />
+        </Inline>
+
+        <Text color="labelQuaternary" size="13pt" weight="bold">
+          {i18n.t(i18n.l.exchange.price_impact.smal_market_try_smaller_amount)}
```

Comment:
we should add some logic for when price impact shows 100% (which means we have no data) and we can also warn them that they should be happy with the quote https://github.com/rainbow-me/rainbow/pull/5635/files#diff-1cd0ecae5857bb98974614524f7c7c124225df7e11fc1b4b22d176dc2f9abce6R571

Reply:
added a follow-up ticket: https://linear.app/rainbow/issue/APP-1366/handle-price-impact-case-where-we-have-no-price-data